pallets/utxo/src/tests.rs

```diff
@@ -740,3 +681,814 @@ proptest! {
+//         }
+//         let tx = Transaction {
+//             inputs: vec![TransactionInput::new_empty(new_utxo_hash.clone())],
+//             outputs: vec![TransactionOutput {
+//                 value: 0,
+//                 destination: Destination::Pubkey(alice_pub_key),
+//                 data: Some(nft_data.clone()),
+//             }],
+//             time_lock: Default::default(),
+//         }
+//         .sign_unchecked(&[new_utxo], 0, &alice_pub_key);
+//         // Submit
+//         assert!(UtxoStore::<Test>::contains_key(H256::from(new_utxo_hash)));
+//         frame_support::assert_err_ignore_postinfo!(
+//             Utxo::spend(Origin::signed(H256::zero()), tx),
+//             "digital data has already been minted"
+//         );
+//     });
+// }
+
+// This macro using for the fast creation and sending a tx
+macro_rules! test_tx {
+    ($data: ident, $checking: tt, $err: expr) => {
+        execute_with_alice(|alice_pub_key| {
+            let (utxo0, input0) = tx_input_gen_no_signature();
+            let output_new = TransactionOutput {
+                value: 1,
+                destination: Destination::Pubkey(alice_pub_key),
+                data: Some($data.clone()),
+            };
+            let tx = Transaction {
+                inputs: vec![input0],
+                outputs: vec![output_new],
+                time_lock: Default::default(),
+            }
+            .sign_unchecked(&[utxo0], 0, &alice_pub_key);
+            let new_utxo_hash = tx.outpoint(0);
+            let (_, init_utxo) = genesis_utxo();
+            // Send
+            assert!(UtxoStore::<Test>::contains_key(H256::from(init_utxo)));
+            // We can check what error we are expecting
+            if stringify!($checking) == "Err" {
+                frame_support::assert_err_ignore_postinfo!(
+                    Utxo::spend(Origin::signed(H256::zero()), tx),
+                    $err
+                );
+                assert!(UtxoStore::<Test>::contains_key(H256::from(init_utxo)));
+                assert!(!UtxoStore::<Test>::contains_key(new_utxo_hash));
+            } else if stringify!($checking) == "Ok" {
+                // We can check is that success
+                assert_ok!(Utxo::spend(Origin::signed(H256::zero()), tx));
+                assert!(!UtxoStore::<Test>::contains_key(H256::from(init_utxo)));
+                assert!(UtxoStore::<Test>::contains_key(new_utxo_hash));
+            }
+        });
+    };
+}
+
+#[test]
+fn test_tokens_issuance_empty_ticker() {
+    // Ticker empty
+    let data = OutputData::TokenIssuanceV1 {
+        token_ticker: vec![],
+        amount_to_issue: 0,
+        number_of_decimals: 0,
+        metadata_uri: vec![],
+    };
+    test_tx!(data, Err, "token ticker can't be empty");
+}
+
+#[test]
+fn test_tokens_issuance_too_big_ticker() {
+    // Ticker too long
+    let data = OutputData::TokenIssuanceV1 {
+        token_ticker: Vec::from([0u8; 10_000]),
+        amount_to_issue: 0,
```

Comment:
Please use valid values for everything else you're not testing against. Don't make the order of validation a part of your test. It's irrelevant.

Reply:
Fixed!